### PR TITLE
fix(qic): reject nested blueprint inputs

### LINKED ISSUE #6823: Standalone \input lines nested inside an atomic TeX environment lose dependency tracking
## Context

Follow-up from #6818 (fix(qic): keep TeX environments atomic).

PR #6818 folds every line of an atomic non-item environment range directly
into one `NonItemBlock` without re-checking each interior line against
`INPUT_LINE_RE`. A `chatgpt-codex-connector[bot]` P1 review comment pointed
out that a standalone `\input{...}` line that happens to fall inside such a
range therefore never gets its `input_payload` set, so `_input_targets`
(the function that walks `NonItemBlock.input_payload` to build the
selected-file closure) never records the dependency, and the referenced
file is silently dropped from both the QIC and TN manifests.

## Mathematical statement

Not applicable — this is a correctness defect in the QICLean/TNLean blueprint
extraction tooling, not a formalization gap.

## Source

- File: `scripts/qic_blueprint_boundary_report.py:357-366` (the atomic-range
  branch builds `current` directly from `source_lines` and flushes it as one
  block, bypassing the `INPUT_LINE_RE.fullmatch` check at line 368 that
  every other line in the loop goes through)
- Claim (Codex review,
  https://github.com/LionSR/TNLean/pull/6818#discussion_r3825819331):
  "When a retained non-item environment contains a standalone `\input{...}`
  line, this path folds the line into an ordinary block whose
  `input_payload` remains `None`, so `_input_targets` never records the
  dependency and the included file is not added to either selected-file
  closure. The simulated manifest can therefore retain the environment and
  its `\input` command while omitting the target `.tex` file, producing a
  broken extracted blueprint."

## Formalization target

Not applicable (tooling fix). In `blueprint_non_item_blocks`, when building
the atomic-range `current` list, still recognize any interior line matching
`INPUT_LINE_RE` and record it (e.g. as a distinct `input_payload` entry
associated with the atomic block, or by rejecting a `\input` line nested
inside a non-item environment as malformed input analogous to the existing
unmatched/mismatched/unterminated checks) so `_input_targets` keeps seeing
it. Add a regression test with a standalone `\input{...}` line placed inside
a `figure`/`center` environment, asserting the target file is still added to
the appropriate selected-file closure.

## Mathematical reference

Not applicable.

## References

- PR: #6818
- Review comment: https://github.com/LionSR/TNLean/pull/6818#discussion_r3825819331
- Tracking issue: #6622 (Perform the atomic history-preserving extraction into QICLean)

### PR DESCRIPTION
**Pull request by LionSR**
## What changed

- reject a standalone `\input{...}` nested inside any parsed TeX environment, including theorem-like item spans
- report the exact file, input line, and enclosing environment range
- retain ordinary top-level router inputs as independently tracked blocks
- treat standard raw-text environments such as `verbatim`, `lstlisting`, and `minted` as opaque while scanning for TeX environments and input commands
- add regressions for non-item nesting, theorem-item nesting, literal verbatim input text, and literal environment tokens in raw text

Rejecting unsupported nested inputs makes the freeze fail closed instead of retaining an input command while silently omitting its target file. Treating raw text as opaque prevents displayed TeX syntax from creating spurious item spans or balance errors. The issue patch does not change either generated manifest relative to exact main.

## Validation

- `PYTHONPATH=scripts python3 scripts/test_qic_blueprint_boundary_report.py` (29 tests)
- `python3 -m py_compile scripts/qic_blueprint_boundary_report.py scripts/test_qic_blueprint_boundary_report.py`
- two byte-identical schema-v4 reports
- generated blueprint and interface manifests are byte-for-byte identical to exact-main generation
- `python3 scripts/check_import_direction.py --root .`
- `python3 scripts/blueprint_lean_sync.py --root . --ci`
- `python3 scripts/check_reader_facing_prose.py --root . --diff-base origin/main`
- `python3 scripts/check_forbidden_lean_tokens.py --base-ref origin/main`
- `python3 scripts/generate_import_aggregators.py --root . --check`
- `python3 scripts/check_tenkz_policy.py`
- `git diff --check`

No Lean or blueprint source changed. Hosted full CI remains authoritative.

Closes #6823. Advances #6622 and #6560.

### CODE REVIEW

**Comment by LionSR** [comment]
@codex review

@claude review

Please review exact head `029d0a358`, especially the fail-closed nested-input behavior and preservation of top-level router inputs.

File: scripts/qic_blueprint_boundary_report.py [hunk 1]
```diff
@@ -285,6 +285,13 @@ def _atomic_non_item_environment_ranges(
                 f"environment at {relative_root}:{start}-{end} contains "
                 "theorem-like item lines"
             )
+        for line_no in range(start, end + 1):
+            cleaned_line = _strip_tex_comments(source_lines[line_no - 1])
+            if INPUT_LINE_RE.fullmatch(cleaned_line) is not None:
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/qic_blueprint_boundary_report.py]: Fixed in `bdd94028d`. Standalone input validation now runs before item-contained ranges are skipped, and a focused theorem-item regression test proves the report fails closed.

[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/qic_blueprint_boundary_report.py]: Fixed in `bdd94028d`. The parser records raw-text ranges for standard verbatim, listings, minted, and comment environments, then ignores apparent input commands on those literal lines. A nested verbatim regression test confirms the displayed command is not treated as a dependency.

**Comment by LionSR** [comment]
@codex review

@claude review

The final exact head is `bdd94028d`; both substantive findings are implemented and resolved.

File: scripts/qic_blueprint_boundary_report.py [hunk 2]
```diff
@@ -261,15 +273,31 @@ def _atomic_non_item_environment_ranges(
                     f"opened {opened} at line {opened_line}, closed {environment}"
                 )
             ranges.append((opened_line, line_no))
+            if opened in RAW_TEXT_ENVIRONMENTS:
+                raw_text_ranges.append((opened_line, line_no))
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/qic_blueprint_boundary_report.py]: Fixed in `2bdd5eecf`. The shared balanced-environment scanner now enters an opaque state at raw-text environment starts and ignores all nested TeX tokens until the matching raw-text end. The theorem-item scanner uses the same raw-line classification, so displayed `\\begin{theorem}` text cannot create a spurious item. Added a regression containing both a matched literal theorem environment and an unmatched literal environment start inside `verbatim`; all 29 tests and the exact-main report/manifests equivalence gates pass.

**Comment by LionSR** [comment]
@codex review

@claude review

Please review exact head `2bdd5eecf02f3b70c4013a1cc3770620c28e26c7`. The raw-text scanner now treats contents as opaque, and the new regression includes both matched and unmatched literal environment tokens.

File: scripts/qic_blueprint_boundary_report.py [hunk 3]
```diff
@@ -241,35 +342,25 @@ def _atomic_non_item_environment_ranges(
...
+    ranges, raw_text_ranges = _balanced_environment_ranges(
+        source_lines, relative_root
+    )
+    raw_text_lines = {
+        line_no
+        for start, end in raw_text_ranges
+        for line_no in range(start, end + 1)
+    }
+    candidates: list[tuple[int, int]] = []
+    for start, end in ranges:
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/qic_blueprint_boundary_report.py:402]: Added a source-wide active-input scan after the raw-text mask is computed. A standalone input covered by the theorem-plus-attached-proof span is now rejected even when it lies outside every balanced environment; a regression test covers the exact intervening-line case.

File: scripts/qic_blueprint_boundary_report.py [hunk 4]
```diff
@@ -141,15 +152,105 @@ def identifier(self) -> str:
...
     records: list[BlueprintEnvironment] = []
     for path in blueprint_content_files(blueprint_src):
         rel = path.relative_to(blueprint_src.parent).as_posix()
         source_lines = path.read_text(errors="replace").splitlines()
+        raw_text_lines = _raw_text_line_numbers(source_lines)
         stack: list[dict[str, object]] = []
         statements: list[dict[str, object]] = []
         for line_no, line in enumerate(source_lines, start=1):
+            if line_no in raw_text_lines:
+                continue
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/qic_blueprint_boundary_report.py:263]: The attached-proof scan now skips every line in the raw-text mask. A regression test includes both matched and unmatched literal proof openings inside verbatim text followed by a real attached proof.

**Comment by LionSR** [comment]
@codex review

Please review the exact head revision 959006050, including the two new boundary-case fixes and their regression tests.

**Comment by LionSR** [comment]
@claude review

Please review the exact head revision 959006050 for correctness, fail-closed behavior, and test coverage.

File: scripts/qic_blueprint_boundary_report.py [hunk 5]
```diff
@@ -141,15 +152,105 @@ def identifier(self) -> str:
...
+    return raw_lines
+
+
 def blueprint_environments(blueprint_src: Path) -> list[BlueprintEnvironment]:
     """Parse theorem-like items, including an attached proof and intervening lines."""
     records: list[BlueprintEnvironment] = []
     for path in blueprint_content_files(blueprint_src):
         rel = path.relative_to(blueprint_src.parent).as_posix()
         source_lines = path.read_text(errors="replace").splitlines()
+        raw_text_lines = _raw_text_line_numbers(source_lines)
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/qic_blueprint_boundary_report.py:258]: Applied the raw-text mask to the complete statement and theorem-plus-proof text before extracting labels, Lean declarations, dependencies, and references. The new regression test displays all four token classes inside verbatim text and confirms that none enters the report.

**Comment by LionSR** [comment]
@codex review

Please review exact head 5098dc32b, including raw-text masking during metadata extraction.

**Comment by LionSR** [comment]
@claude review

Please review exact head 5098dc32b for fail-closed behavior and complete raw-text opacity.
